toolchain/lower/file_context.cpp

```diff
@@ -217,6 +217,16 @@ auto FileContext::BuildFunctionDecl(SemIR::FunctionId function_id,
   // TODO: Consider tracking whether the function has been used, and only
   // lowering it if it's needed.
 
+  // TODO nit: add is_symbolic() to type_id to forward to
+  // type_id.AsConstantId().is_symbolic(). Update call below too.
+  auto get_llvm_type = [&](SemIR::TypeId type_id) -> llvm::Type* {
+    return type_id.has_value() ? (type_id.AsConstantId().is_symbolic()
+                                      ? GetType(SemIR::GetTypeInSpecific(
+                                            sem_ir(), specific_id, type_id))
+                                      : GetType(type_id))
+                               : nullptr;
```

Comment:
Maybe nested conditional operators are a bit confusing/worth writing out, maybe:
```
if (!type_id.has_value()) {
  return nullptr;
}
if (type_id.AsConstantId().is_symbolic()) {
  return GetType(SemIR::GetTypeInSpecific(...));
}
return GetType(type_id);
```
and/or I guess maybe popping the `GetType(` out could be valid:
```
return GetType(type_id.isConstantId().is_symbolic() ? SemIR::GetTypeInSpecific(...) : type_id);
```